script.js

```diff
@@ -1,4 +1,435 @@
+  var programMove = "stone";
+  if (random == 1) {
+    programMove = "scissors";
+  }
+  if (random == 2) {
+    programMove = "paper";
+  }
+  return programMove;
+};
+
+// function to determine when game is a draw where computer's move and players move is the same.
+var drawGame = function (input, computerMove) {
+  return input == computerMove;
+};
+
+// function to determine when player wins in normal mode
+var playerWin = function (input, computerMove) {
+  return (
+    (input == "scissors" && computerMove == "paper") ||
+    (input == "stone" && computerMove == "scissors") ||
+    (input == "paper" && computerMove == "stone")
+  );
+};
+
+// function to determine when player wins in reversed mode
+// created another var so that we do not need to change the number of games and won percentage logic etc
+var playerWinReserved = function (input, computerMove) {
+  return (
+    (input == "stone" && computerMove == "paper") ||
+    (input == "paper" && computerMove == "scissors") ||
+    (input == "scissors" && computerMove == "stone")
+  );
+};
+
+// function to determine when player inputs an invalid guess.
+var invalid = function (input) {
+  return input != "scissors" && input != "stone" && input != "paper";
+};
+
+// function to determine when player loses (this is for Korean mode)
+var playerLose = function (input, computerMove) {
+  return (
+    (input == "stone" && computerMove == "paper") ||
+    (input == "paper" && computerMove == "scissors") ||
+    (input == "scissors" && computerMove == "stone")
+  );
+};
+
+// function to return the logo corresponding to the move.
+var iconLogo = function (look) {
+  var logo = "none";
+  if (look == "scissors" || look == "reversed scrissors") {
+    logo = "✂️";
+  }
+  if (look == "paper" || look == "reversed paper") {
+    logo = "📜";
+  }
+  if (look == "stone" || look == "reversed stone") {
+    logo = "🗿";
+  }
+  return logo;
+};
+
+// ------------------  Main game ---------------------------------
+//-------------------- Main game ---------------------------------
+
+// define main function to enter different game modes and versions
+var main = function (input) {
+  var myOutputValue = "";
+  console.log(currentGameMode);
+  if (input == "refresh") {
+    currentGameMode = "Choose your game";
+    myOutputValue = `Hello ${userName}! Please choose the game mode you'll like to play. <br><br>Type in "Normal", "Reversed", "Korean", "AI"`;
+  } else if (currentGameMode == "enter user name") {
+    myOutputValue = setUserName(input);
+  } else if (currentGameMode == "Choose your game") {
+    myOutputValue = startGame(input);
+  } else if (currentGameMode == "Normal") {
+    myOutputValue = normalGame(input);
+  } else if (currentGameMode == "Reversed") {
+    myOutputValue = reversedGame(input);
+  } else if (currentGameMode == "Korean") {
+    myOutputValue = koreanGame(input);
+  } else if (currentGameMode == "AI") {
+    myOutputValue = aiGame(input);
+  }
```

Comment:
By far, my favourite part of this code, I think it is very clear how you've decomposed the code into the relevant game modes. Keep it up!